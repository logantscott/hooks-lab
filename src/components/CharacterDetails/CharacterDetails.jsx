import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ allies, enemies, _id, photoUrl, name, gender, weapon, profession, position }) => (
  <div id={_id}>
    <img src={photoUrl} alt={name} />
    <h2>{name}</h2>
    <p>Gender: {gender}</p>
    <p>Allies: {allies}</p>
    <p>Enemies: {enemies}</p>
    <p>Weapon: {weapon}</p>
    <p>Profession: {profession}</p>
    <p>Position: {position}</p>
  </div>
);

Character.propTypes = {
  allies: PropTypes.arrayOf(PropTypes.string),
  enemies: PropTypes.arrayOf(PropTypes.string),
  _id: PropTypes.string.isRequired,
  photoUrl: PropTypes.string,
  name: PropTypes.string.isRequired,
  gender: PropTypes.string,
  weapon: PropTypes.string,
  profession: PropTypes.string,
  position: PropTypes.string
};

export default Character;
