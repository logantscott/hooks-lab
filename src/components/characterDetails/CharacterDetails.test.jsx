import React from 'react';
import { shallow } from 'enzyme';
import CharacterDetails from './CharacterDetails';

describe('CharacterDetails component', () => {
  it('renders CharacterDetails', () => {
    const details = {
      _id: '12345',
      name: 'Aang',
      affiliation: 'Air Nomads',
      gender: 'male'
    };
    
    const wrapper = shallow(<CharacterDetails details={details} />);
    expect(wrapper).toMatchSnapshot();
  });
});
