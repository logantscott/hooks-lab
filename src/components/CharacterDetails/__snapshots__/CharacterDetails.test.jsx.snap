// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails component renders CharacterDetails 1`] = `
<Fragment>
  <img
    alt="Aang"
  />
  <h2>
    Aang
  </h2>
  <p>
    Gender: 
    male
  </p>
  <p>
    Allies: 
  </p>
  <p>
    Enemies: 
  </p>
  <p>
    Weapon: 
  </p>
  <p>
    Profession: 
  </p>
  <p>
    Position: 
  </p>
  <p>
    Affiliation: 
    Air Nomads
  </p>
</Fragment>
`;
